src/nvim/charset.c

```diff
@@ -370,7 +370,8 @@ size_t transstr_buf(const char *const s, char *const buf, const size_t len)
   while (*p != NUL && buf_p < buf_e) {
     const size_t l = (size_t)utfc_ptr2len((const char_u *)p);
     if (l > 1) {
-      if (buf_p + l >= buf_e) {
+      //ignore lengths that do not fit the buffer
+      if (buf_p + l > buf_e) {
         break;
       }
       int pcc[MAX_MCO + 2];
```

Comment:
Thats very true, i'll do that.

Reply:
`utfc_ptr2char` says it wants a buffer with space for at least `MAX_MCO+1` chars, and since we put in `&pcc[1]` `pcc[MAX_MCO+1]` seems to be ok. Propably the fix would be 
```
if (i <= MAX_MCO)      /* last composing char must be 0 */                         
  pcc[i] = 0;  
``` 
in `utfc_ptr2char` so the 0 is set even if we found MAX_MCO composing chars.

Reply:
also `utfc_ptr2char` stops at 
```
    for (;; ) {                                                                      
      pcc[i++] = cc;                                                                 
      if (i == MAX_MCO)                                                              
        break;   
```
so pcc only holds MAX_MCO-1 composing chars; or am i wrong?

Reply:
I still dont understand: the loop in `utfc_ptr2char` breaks on `i == MAX_MCO`; if `pcc` is smaller then `MAX_MCO+1` we would write out of bounds, or am i wrong?

Reply:
Yep you are right, i re-read the function more carefully this time. ( should i change the docstring in this PR? )